compiler/semfold.nim

```diff
@@ -637,7 +637,7 @@ proc getConstExpr(m: PSym, n: PNode; g: ModuleGraph): PNode =
       of mLengthArray:
         # It doesn't matter if the argument is const or not for mLengthArray.
         # This fixes bug #544.
-        result = newIntNodeT(lengthOrd(g.config, n.sons[1].typ), n, g)
+        result = newIntNodeT(lengthOrd(g.config, n.sons[1].typ.skipTypes(tyUserTypeClasses)), n, g)
```

Comment:
I think ``lengthOrd`` should do this skipping then?

Reply:
Indeed, I've shuffled some other checks to make all the routines more or less uniform-looking